[{"url":"https://qiita.com/ccaiueo/items/1d26ab3ae254bde883b0","title":"プロジェクトクリーン時のCheckstyleエラー","body":"## 事象\nCheckStyle使用環境下でクリーンした際に<font color=\"red\"> 'Got an exception - expecting xxx, found EOF'</font>エラーが出る\n\n`<module name=\"TreeWalker\"></module>` 間や\n`<module name=\"Checker\"></module>` 間に以下を追加しても直らない\n\n```\n<property name=\"charset\" value=\"UTF-8\"/>\n```\n\n## 原因\nCheckstyleプラグインのバージョンが古い\n\n\n## 解決方法\nCheckStyleのバージョンを上げる\n<br>\nHELP ＞ Install New Software ＞ Add\n![image.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/243523/07c21a11-86bc-f6ea-074c-fdf2fe12e375.png)\n`https://checkstyle.org/eclipse-cs/update`\n<br>\n※インストール時のエラー\nhttps://logroid.blogspot.com/2012/05/eclipse-no-repository-found-containing.html\n上記と同様のエラーが出たが何故か自分はプロキシ設定＆再起動してから実施解消した\n<br>\n\n### 再起動してバージョンが上がっていることを確認\n![image.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/243523/662e3f0e-8b06-a132-80c3-4e0476a273f0.png)\n\n### クリーン再実施\n以下のようなエラーが発生したが、参考URL実施で解消\n![image.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/243523/7f5c8308-3c6e-29c7-8b2c-f15e1c13b195.png)\n【参考】https://qiita.com/ponsuke0531/items/02d3761c83ec2e0681d0\n\n\n\n<br>\nわりと削除したり直したりしないといけなかった（左：修正後）\n![image.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/243523/42712a57-66d0-7646-fccb-1f753b9e5202.png)\n","user":"ccaiueo","created_at":"2021-03-29T14:03:40+09:00","updated_at":"2021-03-29T14:03:40+09:00"},{"url":"https://qiita.com/k3ntar0/items/6193d6b8841132faaf82","title":"Flutter で pod install が失敗する [!] The 'Pods-Runner' target has transitive dependencies that include statically linked binaries","body":"\n## 出力されているエラー\n\n```sh\n$ pod install\nAnalyzing dependencies\ncloud_firestore: Using Firebase SDK version '6.33.0' defined in 'firebase_core'\nfirebase_analytics: Using Firebase SDK version '6.33.0' defined in 'firebase_core'\nfirebase_core: Using Firebase SDK version '6.33.0' defined in 'firebase_core'\nfirebase_messaging: Using Firebase SDK version '6.33.0' defined in 'firebase_core'\nfirebase_remote_config: Using Firebase SDK version '6.33.0' defined in 'firebase_core'\nDownloading dependencies\n[!] The 'Pods-Runner' target has transitive dependencies that include statically linked binaries: (/path-to/ios/Flutter/Flutter.framework)\n```\n\n## 解決方法\nPodfileの `use_frameworks!` をコメントアウトする\n\n```sh\ntarget 'Runner' do\n  👇 コメントアウト\n  # use_frameworks! \n  use_modular_headers!\n\n  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))\nend\n```\n\n以上です\n\ncf. https://github.com/flutter/flutter/issues/20045#issuecomment-409492211\n","user":"k3ntar0","created_at":"2021-03-29T13:57:31+09:00","updated_at":"2021-03-29T13:57:31+09:00"},{"url":"https://qiita.com/toki_k/items/a95229217ec7ea9186c5","title":"Github 初push時のエラー ~ src refspec main does not main any","body":"githubで新しくプロジェクトを作成して、ローカルの物をpushしようとした時に遭遇するエラーの解決方法のメモです。\n\n#手順/エラー内容\n\ngithubで新しくプロジェクトを作成すると以下の表示が見つかります。\n\n```\n…or push an existing repository from the command line\ngit remote add origin git@github.com:.........\ngit branch -M main\ngit push -u origin main\n```\n従って実行すると、以下の様なエラーになってしまいます。\n\n###git branch -M main 時のエラー\n\n```\n$ git branch -M main\nerror: refname refs/heads/master not found\nfatal: Branch rename failed\n```\n\n###git push -u origin main 時のエラー\n\n```\n$ git push -u origin main\nerror: src refspec main does not main any\nerror: failed to push some refs to 'https://github.com/.../.git'\n```\n\n#解決方法\n\n###git branch -M main 時のエラー\ngit checkout -b でブランチを作れます。\n\n```\ngit checkout -b main\n``` \nこれでmainブランチに行けます。\n\n###git push -u origin main 時のエラー\n\ngit add して\n\n```\ngit add .\n```\nコミットします。\n\n```\ngit commit -m 'First Commit'\n```\n\nすると、無事にpushできる様になります。\n\n###参考資料\n[【Git】初プッシュ時にsrc refspec master does not match anyとエラーになる](https://www.mtioutput.com/entry/git-pusherror-refspec)\n","user":"toki_k","created_at":"2021-03-29T13:52:24+09:00","updated_at":"2021-03-29T13:52:24+09:00"},{"url":"https://qiita.com/sakyoyuto/items/7b6d428b78eea208cac4","title":"flutter runでError launching application on iPhone.","body":"#エラー\n\n```\n$ flutter run\n```\nすると\n\n```\nError launching application on iPhone.\n```\nと表示されてしまう。\n実機でのビルド自体はでき、インストールされた状態になるが、上記のエラーが表示されてしまうため、Hot reload等ができない。\n\n#解決法\n\nXcodeを開き、メニューからWindow -> Devices and Simuratorsでウィンドウを開き、ビルドしたい端末を右クリック。\n\nUnpair device\n\nもう一度実行で解決。\n\n（デバイスがBusyだよってのが原因ぽい。）\n","user":"sakyoyuto","created_at":"2021-03-29T13:11:27+09:00","updated_at":"2021-03-29T13:11:27+09:00"},{"url":"https://qiita.com/guy/items/a03151613d1cde6e6591","title":"Rails 6にTailwindCSSを導入して失敗【メモ書き】","body":"\n#背景\n新しいアプリ(ボートフォリオ用)を作成のためTailwind CSSを導入に挑む\n新たな技術を学んでサクサク作れるようになれればよき\n今回もQiitaさんに投稿されている方々にお世話になりました、ありがとうございます\n\n**参考文献**\n[Tailwind on Rails](https://qiita.com/d0ne1s/items/5a2cf50167afd988e9f7)\n[RailsにTailwindCSSを導入](https://qiita.com/kazuya_minei109/items/e76ff2b0af49e69eb441)\n\n##開発環境\n- Ruby 3.0.0\n- Rails 6.1.3.1\n- tailwindcss 2.0.4\n\n##作業工程\nまずは@d0ne1sさんが投稿してくださった「[Tailwind on Rails](https://qiita.com/d0ne1s/items/5a2cf50167afd988e9f7)」の手順で進める\n\n```\n$ yarn add tailwindcss\n$ yarn tailwindcss init\n$ mkdir app/javascript/css\n$ touch app/javascript/css/tailwind.css\n```\n```app/javascript/css/tailwind.css\n@import \"tailwindcss/base\";\n@import \"tailwindcss/components\";\n@import \"tailwindcss/utilities\";\n```\n```app/javascript/packs/application.js\nimport Rails from \"@rails/ujs\"\nimport Turbolinks from \"turbolinks\"\nimport \"channels\"\n\nimport '../css/tailwind.css'; // 追記\n\nRails.start()\nTurbolinks.start()\n```\n```postcss.config.js\nmodule.exports = {\n  plugins: [\n    require('postcss-import'),\n    require('postcss-flexbugs-fixes'),\n    require(\"tailwindcss\"),  // 追記\n    require(\"autoprefixer\"), // 追記\n    require('postcss-preset-env')({\n      autoprefixer: {\n        flexbox: 'no-2009'\n      },\n      stage: 3\n    })\n  ]\n}\n```\n```app/views/layouts/application.html.erb\n// 削除　<%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>\n// 追記　<%= stylesheet_pack_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>\n```\nこれで準備は整ったはず!!\nでは動作確認を行うため準備を・・・・\n\n```app/views/tests/index.html.erb\n<div class=\"max-w-sm mx-auto bg-white shadow-lg rounded-lg overflow-hidden\">\n  <div class=\"sm:flex sm:items-center px-6 py-4\">\n    <img class=\"block mx-auto sm:mx-0 sm:flex-shrink-0 h-16 sm:h-24 rounded-full\" src=\"https://randomuser.me/api/portraits/women/17.jpg\" alt=\"Woman's Face\">\n    <div class=\"mt-4 sm:mt-0 sm:ml-4 text-center sm:text-left\">\n      <p class=\"text-xl leading-tight\">Erin Lindford</p>\n      <p class=\"text-sm leading-tight text-gray-600\">Customer Support Specialist</p>\n      <div class=\"mt-4\">\n        <button class=\"text-purple-500 hover:text-white hover:bg-purple-500 border border-purple-500 text-xs font-semibold rounded-full px-4 py-1 leading-normal\">Message</button>\n      </div>\n    </div>\n  </div>\n</div>\n```\n###動作確認\nでは準備が出来たので「rails s」っと\nあ、うまくいってませんでした・・・・\n![image.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/1215137/a3b9848b-4f96-75ef-cd35-46287c69083e.png)\nうまくいくと下記のサイトに載ってる状態になるみたいです\n[RailsでTailwind CSSを使用する](https://qiita.com/tabakazu/items/4c152de6e9a2c293d1f2)\n\n##作業工程2\n次に@kazuya_minei109さんが投稿してくださった[RailsにTailwindCSSを導入](https://qiita.com/kazuya_minei109/items/e76ff2b0af49e69eb441)の手順も行う\n\nデフォルトで入っているWebpackerを削除して、\n新しいWebpackerを追加を実施\n\n```\n$ yarn remove @rails/webpacker\n$ yarn add rails/webpacker#b6c2180\n```\ngem spring アンインストールを実施\n\n```\n$ bundle exec gem uninstall spring\n```\n\n```\n# gem 'spring'  #念の為にコメントアウト\n```\n```\n$ bundle install\n```\n次に、TailwindsCSS、PostCSS、Autopfrefixer、Tailwindプラグインをインストール\n\n```\nyarn add tailwindcss postcss autoprefixer @tailwindcss/forms @tailwindcss/typography @tailwindcss/aspect-ratio\n```\nこの段階で動作確認を実施\nrails sっと・・・・\n![スクリーンショット 2021-03-29 11.25.20.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/1215137/0e17c7b3-3362-c688-f1b1-aef945dd208f.png)\n\n上手くいってるな・・・・\n\n###少し調査\nここから場合分けで色々試してみた結果\n\n- デフォルトで入っているWebpackerを削除\n- 新しいWebpackerを追加を実施\n- PostCSSのインストール\n\n```\n$ yarn remove @rails/webpacker\n$ yarn add rails/webpacker#b6c2180\n$ yarn add postcss\n```\n上記が必要であった\n下記の\n「tailwindcss」は既にインストール済であったこと\n「autoprefixer」は postcss.config.jsに追記してあったこと\n「@tailwindcss/〜」はtailwindsccのプラグイン各種をインストールすること\nそのため、あまり関係ないかと思います\n\n###さいごに\nPostCSSについても少し調べてみた\n[Rails & Webpacker環境でPostCSSを使う](https://qiita.com/yamashun/items/587e487a885a5291e3a2)\nしかし、理解が追いつきませんでした!!\n少しずつ学んでいくしかないかな\n\n\n\n## 参考文献\n[Tailwind on Rails](https://qiita.com/d0ne1s/items/5a2cf50167afd988e9f7)\n[RailsにTailwindCSSを導入](https://qiita.com/kazuya_minei109/items/e76ff2b0af49e69eb441)\n[RailsでTailwind CSSを使用する](https://qiita.com/tabakazu/items/4c152de6e9a2c293d1f2)\n[Rails & Webpacker環境でPostCSSを使う](https://qiita.com/yamashun/items/587e487a885a5291e3a2)\n","user":"guy","created_at":"2021-03-29T12:34:00+09:00","updated_at":"2021-03-29T12:34:00+09:00"},{"url":"https://qiita.com/kzrashi/items/80ff82d4330d51a398dc","title":"pyenvでインストール時に「BUILD FAILED (OS X 10.15.7 using python-build 20180424)」が発生した場合の対処法","body":"\n# 概要\nmacのpyenvでpythonをインストールしようとした際に、以下のエラーが発生しました。\n以下の手順で動くようになったので、対処法を記載します\n\n## エラー内容\n```zsh\n$ pyenv install 3.5.6\n　:\n　:\n\npython-build: use openssl@1.1 from homebrew\npython-build: use readline from homebrew\nDownloading Python-3.5.6.tar.xz...\n-> https://www.python.org/ftp/python/3.5.6/Python-3.5.6.tar.xz\nInstalling Python-3.5.6...\npython-build: use readline from homebrew\npython-build: use zlib from xcode sdk\n\nBUILD FAILED (OS X 10.15.7 using python-build 20180424)\n\n```\n\n## 対処法\nエラーメッセージに従い、zlibを追加して環境変数に追加することで解決しました\n\n- brewでzlibをインストールする\n\n```zsh\n$ brew install zlib\n　:\n　:\n######################################################################## 100.0%\n==> Pouring zlib-1.2.11.catalina.bottle.tar.gz\n==> Caveats\nzlib is keg-only, which means it was not symlinked into /usr/local,\nbecause macOS already provides this software and installing another version in\nparallel can cause all kinds of trouble.\n\nFor compilers to find zlib you may need to set:\n  export LDFLAGS=\"-L/usr/local/opt/zlib/lib\"\n  export CPPFLAGS=\"-I/usr/local/opt/zlib/include\"\n\nFor pkg-config to find zlib you may need to set:\n  export PKG_CONFIG_PATH=\"/usr/local/opt/zlib/lib/pkgconfig\"\n\n==> Summary\n🍺  /usr/local/Cellar/zlib/1.2.11: 12 files, 376.4KB\n```\n\n- シェルの設定ファイルに加える  \n※zsh以外のシェルを使用している場合、対応する設定ファイルに追記して下さい\n\n```\n$ echo 'export LDFLAGS=\"-L/usr/local/opt/zlib/lib\"' >> ~/.zshrc\n$ echo 'export CPPFLAGS=\"-I/usr/local/opt/zlib/include\"' >> ~/.zshrc\n$ echo 'export PKG_CONFIG_PATH=\"/usr/local/opt/zlib/lib/pkgconfig\"' >> ~/.zshrc\n```\n\n- ターミナルを再起動する  \n  ※　下記のようにsourceコマンドで設定ファイルを再読み込みしても可能\n\n```\n$ source ~/.zshrc\n```\n\n## 参考サイト\nhttps://github.com/aws/aws-elastic-beanstalk-cli-setup/issues/76\n","user":"kzrashi","created_at":"2021-03-29T12:09:25+09:00","updated_at":"2021-03-29T12:09:25+09:00"},{"url":"https://qiita.com/y-ussy/items/b76e1afa5d0a1dc1b6ba","title":"【エラー】自動デプロイの際のエラー【AWS】","body":"# Aws::Sigv4::Errors::MissingCredentialsError\n\nアプリを自動デプロイした際、下記のようなエラーが出現。\n※アプリのサーバには、EC2、unicorn、Nginxを利用\n\n```\nAws::Sigv4::Errors::MissingCredentialsError: Cannot load `Rails.config.active_storage.service`:\nmissing credentials, provide credentials with one of the following options:\n  - :access_key_id and :secret_access_key\n  - :credentials\n  - :credentials_provider\n```\n\n※上記のエラー内容は下記の方法にて確認した。\n （ターミナルにて、EC2にログインし下記コマンドを入力。）\n\n```\n[ec2-user@ip-000-00-0-000 アプリ名]$ cd current\n[ec2-user@ip-000-00-0-000 current]$ cd log\n[ec2-user@ip-000-00-0-000 log]$ cat unicorn.stderr.log\n```\n\n# 結論\nEC2の環境変数にAWSのアクセスキーが設定がされておらず、デプロイ時にActive storageが読み込まれなかった事が原因。\n\n【EC2に環境変数の設定を行う方法】\nEC2にログインした状態で下記コマンドを入力。\n\n```\n[ec2-user@ip-000-00-00-000 ~]$ sudo vim /etc/environment\n```\ni を入力し、入力モードに変更。\n\n```\nAWS_ACCESS_KEY_ID='Access key ID'\nAWS_SECRET_ACCESS_KEY='Secret access key'\n```\n\nアクセスキーの入力が完了したら、「esc」キー　→　「:wq」の順に入力。\n設定した環境変数が反映されているかを確認する為、exitコマンドにてログアウト。\n\n```\n[ec2-user@ip-172-31-23-189 ~]$ exit\n```\n","user":"y-ussy","created_at":"2021-03-29T11:47:56+09:00","updated_at":"2021-03-29T11:47:56+09:00"},{"url":"https://qiita.com/toki_k/items/dbbb77aa0d6ebd96139c","title":"Nest.js/TypeORMでエラー ~ Cannot use import statement outside a module","body":"以下を使っています。\nマイグレーションファイル作成時にエラーが出たので、その解決方法をメモ。\n\n* Nest.js\n* Typeorm\n* postgres\n\n#エラーまで\nマイグレーションファイルを作成しようと以下のコマンドを実行したらエラーが出ました。\n\n```\nnpm run build\nnpx typeorm migration:generate -d src/migrations -n create-item\n```\n\n#エラー内容\nエラー内容は以下の通りです。\n\n```\n..........src/entities/item.entity.ts:1\nimport {\n^^^^^^\n\nSyntaxError: Cannot use import statement outside a module\n```\n\n#解決方法\n\nentitiesとmigrationsの探しに行く場所を\nコンパイル後のjsに変更することでエラーが解消されました。\n\n```ormconfig.json\n{\n    \"type\": \"postgres\",\n    \"host\": \"localhost\",\n    \"port\": 5432,\n    \"username\": \"root\",\n    \"password\": \"password\",\n    \"database\": \"test_db\",\n    \"entities\": [\"src/entities/**/*.ts\"], \n    \"migrations\": [\"src/migrations/**/*.ts\"],\n    \"logging\": true,\n    \"synchronize\": false\n}\n```\n以下の様に編集\n\n```ormconfig.json\n{\n    \"type\": \"postgres\",\n    \"host\": \"localhost\",\n    \"port\": 5432,\n    \"username\": \"root\",\n    \"password\": \"password\",\n    \"database\": \"test_db\",\n    \"entities\": [\"dist/entities/**/*.entity.js\"],\n    \"migrations\": [\"dist/migrations/**/*.js\"],\n    \"logging\": true,\n    \"synchronize\": false\n}\n```\n\n無事に実行されました。\n\n###参考資料\n[Githubのnestのissuesを参考](https://github.com/nestjs/nest/issues/4283)\n[NestJSで簡単なtudoリストを実装するチュートリアル](https://taroosg.io/nestjs-tutorial)\n","user":"toki_k","created_at":"2021-03-29T11:35:12+09:00","updated_at":"2021-03-29T11:37:09+09:00"},{"url":"https://qiita.com/tomo-IR/items/74d1b4762fe62dc82766","title":"【docker】docker-compose build  したらERROR: Service 'web' failed to build : The command '/bin/sh -c bundle install' returned a non-zero code: 7","body":"railsアプリにローカル環境にdockerを導入していました。buildしようと、```docker-compose build```を実行したら、このようなエラーがでました。\n\n```terminal\nYour bundle is locked to mimemagic (0.3.5), but that version could not be found\nin any of the sources listed in your Gemfile. If you haven't changed sources,\nthat means the author of mimemagic (0.3.5) has removed it. You'll need to update\nyour bundle to a version other than mimemagic (0.3.5) that hasn't been removed\nin order to install.\nERROR: Service 'web' failed to build : The command '/bin/sh -c bundle install' returned a non-zero code: 7\n```\n\nどうも、mimemagicというgemが悪さしてそうで、調べてみると0.3.5はyanked（公開停止）されているようでした。\n\n# 対策\ngem mimemagicをアップデートする。\n\n```terminal\n$ bundle update mimemagic\n```\n\n\nちなみにDOCKERFILEはこのように書いていました。\n\n```dockerfile\nFROM ruby:2.6.6\n\nRUN apt-get update -qq && \\\n  apt-get install -y build-essential \\ \n  libpq-dev \\        \n  nodejs      \n\nRUN mkdir /app_name \n\nENV APP_ROOT /app_name \nWORKDIR $APP_ROOT\n\nADD ./Gemfile $APP_ROOT/Gemfile\nADD ./Gemfile.lock $APP_ROOT/Gemfile.lock\n\nRUN gem install bundler \nRUN bundle install\nADD . $APP_ROOT\n\n```\n\n\n# 参考\nhttps://masahiro.me/2021/03/your-bundle-is-locked-to-mimemagic-0-3-5-but-that-version-could-not-be-found/\n\nhttps://qiita.com/Wacci6/items/dc6ffec9b50578d44d6c\n","user":"tomo-IR","created_at":"2021-03-29T10:24:41+09:00","updated_at":"2021-03-29T10:24:41+09:00"},{"url":"https://qiita.com/hann-solo/items/67a7abe008c767b522e1","title":"失敗から学ぶ｜「Appleでサインイン」していたIFTTTの退会","body":"## はじめに\n\nIFTTTに、「Appleでサインイン」を利用してアカウント作成した。退会しようとおもってハマったが、なんとかなった。\n\n## （たぶん）正しいプロセス\n\n1. 「Appleでサインイン」で生成されたパスワードを、なにか任意のものに変える\n1. アカウント削除する\n\n![IMG_4433 (1).jpeg](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/184128/414e20a0-ff23-e053-3460-d13d8013de83.jpeg)\n\n\n## ハマったこと\n\n### `pw:abcdef123456`を変更せずに退会しようとした\n\n理由がよくわかっていないが、下記のようなメッセージがあって、パスワードを変えないといけないようだ\n\n![SS 2021-03-29 9.09.09.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/184128/020f1809-8781-bd78-6a3b-97ec7b116e0c.png)\n\n### キーチェーン.appに、「有効ではない方のパスワード」が登録されてしまった。\n\nこれは自分のミス……\n\n1. 古いパスワード入力（間違い）\n1. 新しいパスワード入力\n1. キーチェーンには新しいパスワードが登録される\n1. パスワード変更が失敗して、IFTTTは古いパスワードを認識している\n\nとなり、キーチェーン以外にパスワードを保管していなかったので、IFTTTの「有効なパスワード」がわからない状態におちいった。\n\n`Alfred`でクリップボード履歴を取っているんだけど、デフォルトでパスワード管理ソフトは「対象外」に設定されていて、クリップボード履歴からも拾えない。\n\n自分のミスと言えばミスなんだけど、キーチェーン.appを利用するときの「よくあるハマりどころ」だよなあ。\n\n### パスワードリセットを、Apple IDのメールアドレスでやろうとした\n\n「Appleでサインイン」で生成される`xyz789@priveterelay.appleid.com`を使わなければならない。\n\n\n## けっきょくやったこと\n\n1. `ID:xyz789@priveterelay.appleid.com`を用いて、パスワードリセット\n1. パスワードを変更\n1. アカウント削除\n\nパスワード変更は不要だったのかもしれない。\nあるいは、パスワードリセットが必要で、このやりかたが正しかったのかもしれない。\nいまとなっては謎である（再度会員登録して確かめるほどでもない……）\n\n\n## 補足\n\nIFTTTは、Gmailに届いたメールをLINEに転送しようとおもってはじめた。Gmailの仕様変更で、IFTTTに対応しなくなっていた。IFTTTをつかう理由がなくなったので退会。\n\nIFTTTはいいサービスだとおもったし、また使いたいと思うけれど、毎日のようにメールが届くのに閉口した。未使用アカウントを残すのも良くないのでサクサクと削除。\n\n\n## 環境\n\nここで書いていることは、下記のバージョンで実施しました。\n\n- Safari バージョン14.0.2 (16610.3.7.1.9)\n- macOS BigSur 11.1\n\n","user":"hann-solo","created_at":"2021-03-29T09:49:31+09:00","updated_at":"2021-03-29T09:52:52+09:00"},{"url":"https://qiita.com/gonza_kato_atsushi/items/bd52462e2a54968ba57f","title":"Rails (V1::Auth::Customers::ConfirmationsController, but didn't (Zeitwerk::NameError))","body":"## railsのプロジェクトをEc2にデプロイしたのち、`rails console -e production`を行うと次にようなエラーが\n\n```console\n[ec2-user@ip-10-0-0-93 room_backend]$ /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader/callbacks.rb:18:in `on_file_autoloaded': expected file /var/www/room_backend/app/controllers/v1/auth/customers/confirmations_controller.rb to define constant V1::Auth::Customers::ConfirmationsController, but didn't (Zeitwerk::NameError)\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:27:in `block in require'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `tap'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `const_get'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `block (2 levels) in eager_load'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:733:in `block in ls'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `foreach'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `ls'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:404:in `block in eager_load'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:393:in `synchronize'\n\tfrom /home/ec2-user/.rbenv/versions/2.6.1/lib/ruby/gems/2.6.0/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:393:in `eager_load'\n以下略\n\n```\n\nエラー名は Zeitwerk::NameError とのこと\nググってみるとスタックオーバーフローにおんなじような記事があったので処置したら治りました\n\nhttps://stackoverflow.com/questions/57277351/rails-6-zeitwerknameerror-doesnt-load-class-from-module\n\n\n## 解決法\n\n```ruby:config/application.rb\nclass Application < Rails::Application\n   config.load_defaults 6.0\n   #下の一行を追加する\n   config.autoloader = :classic\nend\n```\n","user":"gonza_kato_atsushi","created_at":"2021-03-29T09:29:08+09:00","updated_at":"2021-03-29T09:29:08+09:00"},{"url":"https://qiita.com/tukiyo3/items/e9889200883ff3a8a596","title":"Visual Studio 2017 で Assembly が不足しているのを解決する方法","body":"## 現象\n\nvs2015 Express や vs2019 Community ではビルドできるプロジェクトを\nvs2017 WDExpress で開くと、以下参照が見つからない。\nSSDT や NuGetしてみたがイマイチ解決しなかった。\n\n```go\nMicrosoft.Data.Tools.Components\nMicrosfot.Data.Tools.Schema.Sql.UnitTesting\n```\n\n![ng.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/25728/6173fee5-ef05-a339-d397-8a647aa26969.png)\n\n\n## 解決方法 (`.csproj`から削除)\n\nプロジェクトの`.csproj`ファイルを直接編集し、以下2行を削除し解決した。\n\n```diff:プロジェクトの.csproj\n- <Import Project=\"$(SQLDBExtensionsRefPath)\\Microsoft.Data.Tools.Schema.Sql.UnitTesting.targets\" Condition=\"'$(SQLDBExtensionsRefPath)' != ''\" />\n- <Import Project=\"$(MSBuildExtensionsPath32)\\Microsoft\\VisualStudio\\v$(VisualStudioVersion)\\SSDT\\Microsoft.Data.Tools.Schema.Sql.UnitTesting.targets\" Condition=\"'$(SQLDBExtensionsRefPath)' == ''\" />\n```\n\n## 解決方法2 (NuGet)\n\n```go\n// Microsoft.Data.Tools.Components\nInstall-Package Microsoft.Data.Tools.Msbuild -Version 10.0.61707.210\n\n// Microsfot.Data.Tools.Schema.Sql.UnitTesting\nInstall-Package Microsoft.Data.Tools.UnitTest -Version 10.0.60809\n```\n\nとしたが、`Microsfot.Data.Tools.Schema.Sql.UnitTesting`のほうがバージョン不一致になったため、\n`プロジェクトの.csproj`から手で削除した。\n\n* [NuGet Gallery | Microsoft.Data.Tools.Msbuild](https://www.nuget.org/packages/Microsoft.Data.Tools.Msbuild/)\n* [NuGet Gallery | Microsoft.Data.Tools.UnitTest](https://www.nuget.org/packages/Microsoft.Data.Tools.UnitTest/)\n\n\n## 参考\n\n* [データベース単体テストプロジェクトはVisualStudio2017でビルドされません-VisualStudioフィードバック](https://developercommunity.visualstudio.com/t/database-unit-test-project-does-not-build-in-visua/29842)\n","user":"tukiyo3","created_at":"2021-03-29T08:32:24+09:00","updated_at":"2021-03-29T11:05:35+09:00"},{"url":"https://qiita.com/tomo-IR/items/224d33f14561e759dd16","title":"【docker】db:createすると、Plugin caching_sha2_password could not be loaded...のエラーハマった話","body":"# 背景\nrailsで作成中のアプリに、開発環境にDockerを導入していました。\n\n\n# DBをcreateしようとするも下記エラー\n\nコンテナのビルドは問題なく出来たので、次にDBを作成するべくこのコマンドを実行したら、下記のエラーに遭遇しました。\n\n```terminal\n$ docker-compose run web bundle exec rake db:create\n\n\nCreating network \"golfscore_default\" with the default driver\nCreating golfscore_db_1 ... done\nCreating golfscore_web_run ... done\nPlugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory\nCouldn't create 'golfscore_development' database. Please check your configuration.\nrake aborted!\nMysql2::Error::ConnectionError: Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory\n```\n\n# 原因と対策\nMysql 8以降、認証プラグインの仕様が変わったためのようです。\nユーザーのプラグインをmysql_native_passwordに変更していきます。\n\n# docker内のmysqlにログインし、plugin変更\nまずはコンテナ内のMysqlにログインするため、コンテナIDを調べます。\n\n```\n$ docker ps                                            \nCONTAINER ID   IMAGE          COMMAND                  CREATED         STATUS         PORTS                               NAMES\nbcc91696ca70   mysql:8.0.23   \"docker-entrypoint.s…\"   3 minutes ago   Up 3 minutes   33060/tcp, 0.0.0.0:4306->3306/tcp   golfscore_db_1\n```\n\nコンテナIDがわかったので、入ります。\n\n```\n$ docker exec -it bcc91696ca70 bash\n```\n\n入れました。Mysqlに入ります。\n\n```\nroot@bcc91696ca70:/# mysql -uroot -p\n```\n\nMysqlのパスワードを入力します。\n\n```\nEnter password: \nWelcome to the MySQL monitor.  Commands end with ; or \\g.\nYour MySQL connection id is 9\nServer version: 8.0.23 MySQL Community Server - GPL\n\nCopyright (c) 2000, 2021, Oracle and/or its affiliates.\n\nOracle is a registered trademark of Oracle Corporation and/or its\naffiliates. Other names may be trademarks of their respective\nowners.\n\nType 'help;' or '\\h' for help. Type '\\c' to clear the current input statement.\n```\n\nこのコマンドを実行して、現在のプラグインの状況を確認します。\n\n```sql\nmysql> SELECT user, host, plugin FROM mysql.user; \n```\n\nやはり、caching_sha2_passwordになっています。今回はこれが悪さしているようですので、変更していきます。\n![スクリーンショット 2021-03-29 5.53.45.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/693502/1c5dc6f9-ceb1-10de-ad9c-1bc8fe0d08ae.png)\n\n下記、2コマンド実行して、root user部分の2箇所を変更します。\n('password'部分は各自DBのパスワードを入力)\n\n```sql\nmysql> ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';\nQuery OK, 0 rows affected (0.01 sec)\n```\n\n```sql\nmysql> ALTER USER 'root'@'%' IDENTIFIED WITH mysql_native_password BY 'password';\nQuery OK, 0 rows affected (0.01 sec)\n```\n\n変更できたか確認してみます。\n\n```sql\nmysql> SELECT user, host, plugin FROM mysql.user;\n```\n無事、mysql_native_passwordに変更されています。\n![スクリーンショット 2021-03-29 5.53.35.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/693502/d4e93dfb-ccba-835f-34a0-42abdf6ddc00.png)\n\n\n再度、DBを作成し、無事に成功しました。\n\n```terminal\n$ docker-compose run web bundle exec rake db:create    \nCreating golfscore_web_run ... done\nCreated database 'golfscore_development'\nCreated database 'golfscore_test'\n```\n\n\n# 参考\nhttps://yamasey712.hatenablog.com/entry/2019/08/28/213507\n\nhttps://qiita.com/majorboy/items/9fbfc78fc7bbc1f35e77\n\nhttps://qiita.com/shungo_m/items/5363c16b698ef6310014\n","user":"tomo-IR","created_at":"2021-03-29T06:14:57+09:00","updated_at":"2021-03-29T06:14:57+09:00"},{"url":"https://qiita.com/kyokucho1989/items/3001d3bf4c67511be4d8","title":"thisなきJS エラーあり","body":"# はじめに\n　JSでTODOアプリをつくり、そのやり方を教材として書き起こしていた。教材の仕様に沿ってアロー関数をつかったらうまく動作しなくなった。なんだこれは。少しだけ考えた。\n\n# 問題\n　問題が発生したのはTODOの内容をクリックして、既存のタスクを編集する箇所。今までのfunctionではOKだったが、アロー関数にするとなんだかおかしい。\n\n\n# コード\n　問題点を明らかにするためにその箇所だけを抜き出して別にコードを書いた。「テスト」をクリックするとアロー関数のやつが実行されて、「テスト２」だと普通の関数が実行される。\n　テストをクリックして、入力フォームのカーソルを外すと\n\n　「Uncaught TypeError: Cannot read property 'classList' of null」\n\n　というエラーが出てくる。\n　\n\n```html:test.html\n<!DOCTYPE html>\n<html lang=\"ja\">\n<head>\n    <meta charset=\"utf-8\"/>\n    <link rel =\"stylesheet\" href=\"todo.css\">\n    <title>イベントハンドラテスト</title>\n</head>\n<body>\n  <header>\n    <h1>TODOテスト</h1>\n  </header>\n  <div class=\"container\">\n    <div class=\"todo-container\">\n      <ul class=\"todo-list\">\n        <li>\n          <span class=\"todo-content1\">テスト</span>\n        </li>\n        <li>\n          <span class=\"todo-content2\">テスト2</span>\n        </li>\n      </ul>\n    </div>\n  </div>\n    <script src = \"test.js\"></script>\n</body>\n</html> \n```\n\n```javascript\nconst editTodo = (e) => {\n  let itemToEdit = e.target;\n  if(!itemToEdit.classList.contains('on')) {\n    itemToEdit.classList.add('on');\n    let contentBeforeEdit = itemToEdit.textContent;\n    itemToEdit.innerHTML = '<input type=\"text\" class=\"editbox1\" value=\"'+contentBeforeEdit+'\" />';\n    const editContent1 = document.querySelector('.editbox1');\n    \n    const saveTodoContent  = (e) => {\n        let itemToSave = e.target;\n        itemToSave.parentNode.classList.remove('on');\n        let txtvalue = itemToSave.value;\n        if (txtvalue ==''){\n         txtvalue = itemToSave.defaultValue;\n        }\n        itemToSave.parentNode.innerHTML = txtvalue;\n        \n    }\n    editContent1.addEventListener('blur',saveTodoContent);\n  }\n}\n\nfunction editTodo2(e) {\n  if(!this.classList.contains(\"on\")) {\n    this.classList.add(\"on\");\n    let contentBeforeEdit = this.textContent\n    this.innerHTML = '<input type=\"text\" class=\"editbox2\" value=\"'+contentBeforeEdit+'\" />'\n    const editContent2 = document.querySelector(\".editbox2\")\n\n    let saveTodoContent = function(){\n      this.parentNode.classList.remove('on')\n      let txtvalue = this.value\n      if (txtvalue ==\"\"){\n       txtvalue = this.defaultValue\n      }\n      this.parentNode.innerHTML = txtvalue\n    \n    }\n    editContent2.addEventListener(\"blur\",saveTodoContent)\n  }\n}\n\n//Select DOM\nconst todoContent = document.querySelector('.todo-content1');\ntodoContent.addEventListener('click', editTodo);\nconst todoContent2 = document.querySelector('.todo-content2');\ntodoContent2.addEventListener('click', editTodo2);\n\n```\n\n# 問題を追う\n\n　カーソルを外した時に「classListがnullですよ」というエラーが発生する。ちょうどこの行である。\n\n```javascript\n const saveTodoContent  = (e) => {\n//\n      itemToSave.parentNode.classList.remove('on');\n//\n```\n\n　Chromeのデベロッパーモードで少しずつ検証していった。そうするとカーソルを外す時に実行されるイベント`saveTodoContent`が２回実行されていることがわかった。そして２回目の実行時にエラーが発生していた。なぜだ。\n\n　だいたいはthisが原因だった。\n\n# 原因\n　この機能では項目の状態を知る必要がある。つまり「タスクが現在入力モードなのか否か」ということを知りたいのだ。コード上ではclasslistにonというclassを付与したり外したりしてそれを操作している。\n　今まではその検知にthisを用いていたが、今回アロー関数に書き換えたことによってthisが使えなくなり、別の記法で書き換えた。そこがよくなかったのだ。\n\n　ここである。\n\n```javascript:変更前\nfunction editTodo2(e) {\n  if(!this.classList.contains(\"on\")) {\n// ..　クリックときに編集モードでなければ以下を実行\n```\n\n```javascript:変更後\nconst editTodo = (e) => {\n  let itemToEdit = e.target;\n  if(!itemToEdit.classList.contains('on')) {\n// ..　クリックときに編集モードでなければ以下を実行\n```\n\n# this/e.target\n\nthisとe.targetはなにが違うのか。thisは場面によってさまざまに容態を変化させるが、ここでは以下の記事を参考にするならば「関数を呼び出している元のオブジェクト」である。\n\nhttps://qiita.com/takkyun/items/c6e2f2cf25327299cf03\n\n　e.targetとはクリックしたときの物体である。テキスト文字であったり、入力フォームであったりする。\n\n　thisの中身はタスクが編集モードか否かでも変わらない。しかしe.targetは中身が変わる。ためしにconsole.logで読んでみよう。\n\n\n```javascript\nconst editTodo = (e) => {\n  let itemToEdit = e.target;\n  console.log(e.target);\n  console.log(this);\n\n//\n\n\nfunction editTodo2(e) {\n  console.log(e.target);\n  console.log(this);\n```\n　\n![スクリーンショット 2021-03-29 5.25.11.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/104211/f239f2ae-37e3-a5c3-7671-efe6c4184489.png)\n\n上4行はアロー関数のほう。下はいままでの関数だ。たしかになんか違っていた。\n\n# どうしたか\n\n　アロー関数にはイベントが実行される条件を増やした。\n\n```javascript\nconst editTodo = (e) => {\n  let itemToEdit = e.target;\n  if(!itemToEdit.classList.contains('todo-content1')){\n    return;\n  }\n  if(!itemToEdit.classList.contains('on')) {\n//....\n\n```\n\n　これで問題は解決した。\n\n\n# おわりに\n\n　文章がまとまらないままこれを書き出してしまった。それでもどこかに残しておかないと、後々自分が困ってしまう。\n\n　もしかしたら追記するかもしれないししないかもしれない。\n\n　\n","user":"kyokucho1989","created_at":"2021-03-29T05:30:32+09:00","updated_at":"2021-03-29T05:30:32+09:00"}]